### PR TITLE
Swift 3! 🎉

### PR DESCRIPTION
**Pull request by jlandon**
Given that we're about a week out from official Xcode 8 and Swift 3 releases, I figured it'd be a good idea to begin reviewing the changes I've made over the summer to support Swift 3 🙂

I've gone through the codebase and feel like all the changes makes sense, but definitely feel free to point out any API changes that you think could be better! There are obviously a _ton_ of changes, so feel free to review at your leisure, but it'd be great to get this merged and released next week on the 7th!

### CODE REVIEW

**Comment by krider2010** [comment]
🚢  - I want to take it into production!

File: Sources/UITableView+Extensions.swift [hunk 1]
```diff
@@ -136,86 +146,95 @@ extension UITableView {
...
         } else {
-            return NSIndexPath(forRow: indexPath.row + 1, inSection: indexPath.section)
+            return IndexPath(row: indexPath.row + 1, section: indexPath.section)
         }
     }
     
     
     /**
-     When "sequentially" traversing a `UITableView`, what's the previous `NSIndexPath` before the given `NSIndexPath`.
+     When "sequentially" traversing a `UITableView`, what's the previous `IndexPath` before the given `IndexPath`.
```
Comment (philtr) [inline Sources/UITableView+Extensions.swift]: 👌

**Comment by courtsimas** [comment]
After a close review, I give my 👍

**Comment by hsoi** [comment]
I say let's go for it. If there are problems, we'll sort 'em out as we encounter them.

**Pull request merged**